CMakeLists.txt

```diff
@@ -60,6 +60,7 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_CXX_STANDARD 17)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=deprecated-declarations" )
```

Comment:
What caused this? If it is not related to this change it should be dealt with separately. Also consider that only `Debug` builds get `-Werror`, so if this is really needed then it should be put in `CMAKE_CXX_FLAGS_DEBUG`